lib/mirage/mirage.mli

```diff
@@ -15,19 +15,103 @@
  * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  *)
 
-(** Mirage combinators.
+(** {e Release %%VERSION%%} *)
+
+(** {1 What is MirageOS?}
+
+    MirageOS is a library operating system that can build standalone unikernels
+    on various platforms. More precisely, the architecture can be divided into:
+
+    - {e operating system libraries} that implement kernel and protocol
+      functionality, ranging from low-level network card drivers to a full
+      reimplementation of the TLS protocol, through to a reimplementation of the
+      Git protocol to store versioned data.
+
+    - A set of {e typed signatures} to make sure these libraries are consistent
+      and can interoperate. As all the library are almost all pure OCaml code,
+      we have defined {e a set of OCaml module types} that encode these
+      conventions in a statically enforcable way. We make no compatibility
+      guarantees at the C level, but compile those on a best-effort basis.
+
+    - Finally, MirageOS is also a {e metaprogramming compiler} that generates
+      OCaml code. It takes as input: the OCaml source code of a program and all
+      of its dependencies, the full description of the deployment target,
+      including configuration values (like the HTTP port to listen too, or the
+      private key or the service being deployed). The `mirage`CLI tool uses all
+      of these to {e generate a executable unikernel}: a specialised binary
+      artefact containing only the code what is needed to run on the given
+      deployment platform and no more.
+
+    It is possible to write high-level MirageOS applications, such as HTTPS,
+    email or CalDAV servers which can be deployed on very heterogenous and
+    embedded platforms by changing only a few compilation parameters. The
+    supported platforms range from ESP32 micro-controllers, minimal virtual
```

Comment:
I am not sure what you are targeting here -- if it is a description of the MirageOS tool (as released in 3.x series), I'd stay to the supported and implemented targets (I don't think the ESP32 target is merged and released)

Reply:
fixed